rakelib/pkg.rake

```diff
@@ -75,9 +75,13 @@ namespace :pkg do
       mkdir_p ".cargo"
       sh "cargo vendor --versioned-dirs --locked #{vendor_dir} >> #{cargo_config_path} 2>/dev/null"
 
-      final_gemspec.files += dirglob("**/.cargo/**/*")
-      final_gemspec.files += dirglob("./#{vendor_dir}/**/*")
-      final_gemspec.files.reject! { |f| File.directory?(f) }
+      vendor_files = dirglob("./#{vendor_dir}/**/*").reject { |f| File.directory?(f) }
+      # Ensure that all vendor files have the right read permissions,
+      # which are needed to build the gem.
+      # The permissions that we want are `.rw-r--r--`
```

Comment:
Nit: we want at least readable by all (`perm & 444 == 444`), it doesn't need to be exactly `rw-r--r--`